pkg/controller/servicemesh/memberroll/controller_test.go

```diff
@@ -190,13 +190,77 @@ func TestReconcileFailsIfListingNamespacesFails(t *testing.T) {
 	test.AssertNumberOfWriteActions(t, tracker.Actions(), 0)
 }
 
+func TestReconcileReconcilesAfterOperatorUpgradeFromV1_0(t *testing.T) {
+	roll := newMemberRoll()
+	addOwnerReference(roll)
+	roll.Spec.Members = []string{appNamespace}
+	roll.Status.ConfiguredMembers = []string{appNamespace}
+	roll.ObjectMeta.Generation = 1
+	roll.Status.ObservedGeneration = 1
+	controlPlane := markControlPlaneReconciled(newControlPlane("v1.0"))
+	roll.Status.ServiceMeshGeneration = controlPlane.Status.ObservedGeneration
+	roll.Status.ServiceMeshReconciledVersion = "1.0.0-1"
+	namespace := newAppNamespace()
+	common.SetLabel(namespace, common.MemberOfKey, controlPlaneNamespace)
+	meshRoleBinding := newMeshRoleBinding()
+	appRoleBinding := newMeshRoleBinding()
+	appRoleBinding.SetNamespace(appNamespace)
+	common.SetLabel(appRoleBinding, common.MemberOfKey, controlPlaneNamespace)
+	nad := createNAD("istio-cni", appNamespace, controlPlaneNamespace)
+
+	cl, tracker, r := createClientAndReconciler(t, roll, controlPlane, namespace, meshRoleBinding, appRoleBinding, nad)
+	tracker.AddReactor(assertNADNotDeleted(t))
+	tracker.AddReactor(assertRBNotCreated(t))
+	common.IsCNIEnabled = true // TODO: this is a global variable; we should get rid of it, because we can't parallelize tests because of it
+
+	assert.Equals(roll.Status.ServiceMeshReconciledVersion != controlPlane.Status.GetReconciledVersion(), true, "Unexpected Status.ServiceMeshReconciledVersion in SMMR already matches SMCP reconciled version", t)
+
+	assertReconcileSucceeds(r, t)
+
+	updatedRoll := test.GetUpdatedObject(cl, roll.ObjectMeta, &maistra.ServiceMeshMemberRoll{}).(*maistra.ServiceMeshMemberRoll)
+	assert.DeepEquals(updatedRoll.Status.ConfiguredMembers, []string{appNamespace}, "Unexpected Status.ConfiguredMembers in SMMR", t)
+	assert.Equals(updatedRoll.Status.ServiceMeshGeneration, controlPlane.Status.ObservedGeneration, "Unexpected Status.ServiceMeshGeneration in SMMR", t)
+	assert.Equals(updatedRoll.Status.ServiceMeshReconciledVersion, controlPlane.Status.GetReconciledVersion(), "Unexpected Status.ServiceMeshReconciledVersion in SMMR", t)
+	assert.Equals(updatedRoll.Status.MeshVersion, common.LegacyMaistraVersion, "MemberRoll.Status.MeshVersion should have been updated to v1.0 (from empty string)", t)
+
+	meshNetAttachDefName := "istio-cni"
+	assertNamespaceReconciled(t, cl, appNamespace, controlPlaneNamespace, meshNetAttachDefName, []rbac.RoleBinding{*meshRoleBinding})
+}
+
+func TestReconcileReconcilesAddedMemberV1_0(t *testing.T) {
```

Comment:
Since this test is almost identical to `TestReconcileReconcilesAddedMember`, it would make sense to use subtests instead of a separate test.

Reply:
Will convert.